pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+              variables are extracted for the repository including
+              the host and disable_pull_operations attributes.
+
+            repository(:class:`~pywbem_mock.BaseRepository` or derived from `~pywbem_mock.BaseRepository`):
+             class defining the CIM repository that stores CIM objects.
+        """  # noqa: E501
+        # pylint: enable=line-too-long
+
+        #   Save required attributes from the connection object
+        self.host = conn.host
+        self.disable_pull_operations = None
+
+        # Implementation of the CIM object repository that is the data store
+        # for CIM classes, CIM instances, CIM qualifier declarations and
+        # CIM methods.
+        # See :py:module:`pywbem_mock/baserepository` for a description of
+        # the repository interface.
+        self.cimrepository = cimrepository
+
+        # Open Pull Contexts. The key for each context is an enumeration
+        # context id.  The data is the namespace for the pull sequence,
+        # the list of remaining instances/names to be returned with subsequent
+        # pull operations. Any enumeration context in this list is still open.
+        self.enumeration_contexts = {}
+
+    def __repr__(self):
+        return _format(
+            "MainProvider("
+            "cimrepository={s.cimrepository}, "
+            "host={s.host}, "
+            "disable_pull_operations={s.disable_pull_operations})",
+            s=self)
+
+    @property
+    def disable_pull_operations(self):
+        """
+        ::`allow`:
+          Boolean Flag to set option to disable the execution of the open and
+          pull operation request handlers in the mock CIM repository. This
+          emulates the characteristic in some CIM servers that did not
+          implement pull operations. The default is to allow pull operations.
+          All pull operations requests may be forbidden from executing by
+          setting disable_pull_operations to True.
+
+          This attribute is settable. For details, see the description of the
+          same-named init parameter of
+          :class:`this class <pywbem.FakedWBEMConnection>`.
+        """
+        return self._disable_pull_operations
+
+    @disable_pull_operations.setter
+    def disable_pull_operations(self, disable):
+        """Setter method; for a description see the getter method."""
+        # Attribute will always be boolean
+        if disable is None:
+            disable = False
+        if isinstance(disable, bool):
+            # pylint: disable=attribute-defined-outside-init
+            self._disable_pull_operations = disable
+        else:
+            raise ValueError(
+                _format('Invalid type for disable_pull_operations: {0!A}, '
+                        'must be a boolean', disable))
+
+    @property
+    def namespaces(self):
+        """
+        Return the namespaces that exist in the datastore
```

Comment:
Change "in the datastore" to "in the CIM repository".

Reply:
DONE